src/back/glsl/mod.rs

```diff
@@ -570,27 +532,19 @@ impl<'a, W: Write> Writer<'a, W> {
 
         let ep_info = self.info.get_entry_point(self.entry_point_idx as usize);
 
-        // Write all structs
+        // Write struct types.
         //
-        // This are always ordered because of the IR is structured in a way that you can't make a
-        // struct without adding all of it's members first
+        // This are always ordered because the IR is structured in a way that
+        // you can't make a struct without adding all of its members first.
         for (handle, ty) in self.module.types.iter() {
             if let TypeInner::Struct { ref members, .. } = ty.inner {
-                let generate_struct = match GlobalTypeKind::new(self.module, handle) {
-                    GlobalTypeKind::WrappedStruct => true,
-                    // If it's a global non-wrapped struct, it will be printed
-                    // with the corresponding global variable.
-                    GlobalTypeKind::Unsized(_) => false,
-                    GlobalTypeKind::Other => {
-                        let used_by_global =
-                            self.module.global_variables.iter().any(|(vh, var)| {
-                                !ep_info[vh].is_empty() && var.space.is_buffer() && var.ty == handle
-                            });
-                        // If not used by a global, it's safe to just spew it here
-                        !used_by_global
-                    }
```

Comment:
Where did this code go? FIY it's needed because some systems might not support things like dynamic arrays and this code prevents it from being generated if it isn't requested.

Reply:
That's not what this code says. It says, if the struct type is *not* used by any global, then the backend should write its definition. So it will cause *unused* structs to be written.

In any case, this code isn't actually reachable, because `GlobalTypeKind::new` will never return `Other` for a `Struct` type.

Reply:
🤦🏻 please ignore me